src/recommendations/RecommendationsList.jsx

```diff
@@ -15,6 +16,17 @@ const RecommendationsList = (props) => {
     title, recommendations, userId, setSelectedRecommendationsType, selectedRecommendationsType,
   } = props;
 
+  useEffect(() => {
+    if (recommendations.length) {
+      const viewedProductsList = recommendations.map((product) => ({
+        product_key: product.uuid,
+        product_line: product.cardType,
+        product_source: product.productSource.name,
+      }));
+      trackRecommendationsViewed(viewedProductsList, false, userId);
+    }
+  }, [recommendations, userId]);
```

Comment:
`selectedRecommendationsType` should also be part of this useEffect. otherwise on tab change, this event will not be fired.

Reply:
Addressed.